apps/android/app/src/main/java/chat/simplex/app/views/helpers/Modifiers.kt

```diff
@@ -0,0 +1,17 @@
+package chat.simplex.app.views.helpers
+
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.layout
+
+fun Modifier.badgeLayout() =
+  layout { measurable, constraints ->
+    val placeable = measurable.measure(constraints)
+
+    // based on the expectation of only one line of text
+    val minPadding = placeable.height / 4
+
+    val width = maxOf(placeable.width + minPadding, placeable.height)
+    layout(width, placeable.height) {
+      placeable.place((width - placeable.width) / 2, 0)
+    }
+  }
```

Comment:
Copied from https://stackoverflow.com/questions/69987940/jetpack-compose-custom-badge-with-dynamic-text-size-with-circleshape-is-not-draw